snap/info.go

```diff
@@ -643,28 +644,69 @@ func (s *Info) CommonDataSaveDir() string {
 	return CommonDataSaveDir(s.InstanceName())
 }
 
+// Get home directories from system-params file.
+// Home directories are returned as a ',' seperated string
+func HomeDirsSysParams(rootDir string) (string, error) {
+
+	ssp, err := sysparams.Open(rootDir)
+	if err != nil {
+		return "", err
+	}
+
+	return ssp.Homedirs, nil
+}
+
 // DataHomeGlob returns the globbing expression for the snap directories in use
-func DataHomeGlob(opts *dirs.SnapDirOptions) string {
+func DataHomeGlob(opts *dirs.SnapDirOptions) []string {
+
 	if opts == nil {
 		opts = &dirs.SnapDirOptions{}
 	}
 
-	if opts.HiddenSnapDataDir {
-		return dirs.HiddenSnapDataHomeGlob
+	homedirs, err := HomeDirsSysParams(dirs.GlobalRootDir)
+
+	// if we cannot read the file or if the homedirs value is empty we use dirs defaults
```

Comment:
I think we need to append default dirs regardless of `system.homedirs`.
@pedronis what do you think?
> The original /home location remains valid but it is no longer a requirement that directories be stored there.
https://snapcraft.io/docs/home-outside-home

Reply:
Hello @ZeyadYasser, thank you for reviewing the PR!
I also though if I should add `/home` to `homedirs`, however this will cause a cases of double copying/deleting when upgrading/removing a snap if the user wrote something like `sudo snap set system homedirs=/my/new/home,/home` .
Obviously I can write a check within the function for this scenario, but I think this check should be done while the `snap set system homedirs=` command is being ran and always add `/home` by default to the `system-params` file if not specified directly by the user.
I can add this check inside the `updateHomedirsConfig` function in `overlord/configstate/configcore/homedirs.go`. 
Doing so will avoid duplicate code for each (current or future) function reading the `system-params` file and avoid any potential regressions in case a function did not implement this check. 
I would really appreciate any feedback on this approach.

Reply:
I understand the appeal of that approach but I don't think we can do it that way as it requires also changes to snap-confine which is the other consumer of system-params